### PR TITLE
PyQt GUI

### PR DESCRIPTION
**Pull request by IguanasInPyjamas**
So, Originally I wanted this front end to be a placeholder while Soldat improves my previous UI. However, I really like the clean and simple look of this GUI. It seems to suit the aesthetic of anime-dl in my eyes. So, I'm probably going to pursue this GUI for the foreseeable future. 

Only one extra dependency - PyQt5. You're gonna need a dependency no matter what we do if we want to not build a GUI environment from scratch. 

I've added all the "normie" features, episode slices still works, Nyaa still works even with the episodes options.

Ezdl has not been included yet, I'm going to implement a tick box for ezdl. However, is it even really necessary in this GUI.

Theming options will be coming soon, devs and collabs will have the option to give a specific theme for them.

I personally feel like a GUI like this is superior to the cloudfare port alternative. Firstly, the code is simple enough for other people to maintain if I ever leave. Secondly, the *Netflix style* doesn't really suit anime-dl especially when we don't have a native player. Another advantage is the use of actual anime-dl commands. 

I haven't added a GUI command to the anime cli, I would suggest people just adding the script to their hotbar etc. as that allows the normies to just use a GUI.

This PR is a draft, mainly because I need to add comments especially around the threading component, take feedback on theming options from other devs and feature additions.

Any feedback would be appreciated amigos.

### CODE REVIEW

**Comment by ArjixGamer** [comment]
add PyQt5 as a requirement in setup.py

**Comment by IguanasInPyjamas** [comment]
Good catch.

**Comment by IguanasInPyjamas** [comment]
Issue -> PySmartDL breaks the internal downloader, was fine before I updated to that... something is going wrong with the backwards compatibility.

**Comment by Blatzar** [comment]
PySmartDL replaced the old internal downloader, how are you accesing the internal?

**Comment by ArjixGamer** [comment]
image

**Comment by Blatzar** [comment]
It cannot be accessed in that way, using episode.download() if there's no external is how it's done.

image

**Comment by Blatzar** [comment]
Feedback:
"..." to select download dir isn't very intuitive imo, might be placeholder.
A play button would be nice. I could probably contribute with those though.

File: anime_downloader/config.py [hunk 1]
```diff
@@ -9,7 +9,7 @@
 DEFAULT_CONFIG = {
     'dl': {
         'url': False,
-        'player': None,
+        'player': 'mpv',
```
[the remark this answers is not in the document]
Comment (Blatzar) [reply anime_downloader/config.py:12]: https://github.com/IguanasInPyjamas/anime-downloader/pull/5
for fix

**Comment by IguanasInPyjamas** [comment]
Issue -> Terminating the thread when a download is activating either doesn't stop the download or full errors out the app. We need some handling for this

**Comment by Blatzar** [comment]
Features needed:
Default theme in config
Default provider in config

Bug: Spamming the download button will start multiple downloads and eventually kill the gui while the threads are running in the background

**Comment by AbdullahM0hamed** [comment]
> we only have like 4 themes

Probably better to add theme settings in which the user can select the colours of certain aspects of the UI as opposed to adding more

**Comment by IguanasInPyjamas** [comment]
@AbdullahM0hamed feel free to autoPEP8 this for me :)

File: setup.py [hunk 2]
```diff
@@ -30,6 +30,7 @@
         'cfscrape>=2.0.5',
         'requests-cache>=0.4.13',
         'tabulate>=0.8.3',
+        'PyQt5>=5.15.1',
```
Comment (AbdullahM0hamed) [inline setup.py:33]: Pyqt doesn't install on android (at least via termux) - so this breaks anime-downloader on android, and it's only needed by people who will use the GUI - perhaps move this to `extras_require`?

**Review by AbdullahM0hamed — CHANGES_REQUESTED** [review]

File: setup.py (hunk 2 above)
Comment (IguanasInPyjamas) [reply setup.py:33]: That's a great idea. I'll do that

**Review by AbdullahM0hamed — APPROVED** [review]

**Pull request merged**